pkg/minio/minio.go

```diff
@@ -117,6 +102,42 @@ func NewClient(bucket *sourcev1.Bucket, opts ...Option) (*MinioClient, error) {
 		})
 	}
 
+	switch bucketProvider := bucket.Spec.Provider; {
+	case secret != nil:
+		var accessKey, secretKey string
+		if k, ok := secret.Data["accesskey"]; ok {
+			accessKey = string(k)
+		}
+		if k, ok := secret.Data["secretkey"]; ok {
+			secretKey = string(k)
+		}
+		if accessKey != "" && secretKey != "" {
+			minioOpts.Creds = credentials.NewStaticV4(accessKey, secretKey, "")
+		}
+	case bucketProvider == sourcev1.AmazonBucketProvider:
+		stsEndpoint := ""
+		if sts := bucket.Spec.STS; sts != nil {
+			if sts.Provider != sourcev1.AmazonSTSProvider {
+				return nil, fmt.Errorf("STS provider '%s' is not supported for Bucket provider '%s'",
+					sts.Provider, sourcev1.AmazonBucketProvider)
+			}
+			stsEndpoint = sts.Endpoint
+		}
+
+		creds := credentials.NewIAM(stsEndpoint)
+		if len(transportOpts) > 0 {
+			transport := http.DefaultTransport.(*http.Transport).Clone()
+			for _, opt := range transportOpts {
```

Comment:
Does it make sense to use the TLS config here besides proxy?

Reply:
If a custom CA is specified, I expect for both endpoints to use, we're not going to introduce a new tlsSecretRef for STS so I think it's Ok to reuse the CA for now.

Reply:
Indeed. I'm gonna remove it.

Reply:
Fixed, thanks!

Reply:
As discussed in the dev meeting, we will address this for the future `ldap` STS provider.